client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,181 @@
-jest.mock( 'calypso/blocks/dismissible-card', () => {
-	return function DismissibleCard() {
-		return null;
-	};
-} );
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import preferencesReducer from 'calypso/state/preferences/reducer';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
+import { Banner } from '../index';
 
-jest.mock( 'calypso/lib/analytics/track-component-view', () => {
-	return function TrackComponentView() {
-		return null;
-	};
-} );
+jest.mock( 'calypso/my-sites/plan-price', () => ( props ) => (
+	<div data-testid="banner-price-plan-mock">{ props.rawPrice }</div>
```

Comment:
As pointed out in the earlier comment, feel free to pass `original` and/or `discounted` in a `class`, or any other attribute here, in order to be able to test it as well.